gpbft/gpbft.go

```diff
@@ -211,7 +211,8 @@ type instance struct {
 	// independently of protocol phases/rounds.
 	decision *quorumState
 	// tracer traces logic logs for debugging and simulation purposes.
-	tracer Tracer
+	tracer   Tracer
+	progress ProgressObserver
```

Comment:
I.e.:

1. Have the instance notify the host (everything but some small crypto stuff goes through the host).
2. The host notifies the validator (and anything else it might want to notify (e.g., lotus?).

Reply:
As in make it part of the Host interface?

Reply:
Yeah, that's the idea.

Reply:
How would this work for validation?

`MessageValidator` is part of the `Receiver` interface which is implemented by `Participant`, not `Host`.

I guess to make this work both `Participant` and  `Host` need to be `ProgressObserver`s. 🤔

Reply:
Discussed 1:1; there is a body of work around reducing mutex usage across `Host` and `Participant` which is worth doing but in a separate PR.

Reply:
Captured https://github.com/filecoin-project/go-f3/issues/658.